packages/sil-governance/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx

```diff
@@ -33,8 +36,11 @@ const Buttons = ({ id, vote }: Props): JSX.Element => {
       const connection = await getBnsConnection();
       const voteTx = await governor.buildVoteTx(id, currentVote);
 
-      await sendSignAndPostRequest(connection, voteTx);
-      setPreviousVote(currentVote);
+      const transactionId = await sendSignAndPostRequest(connection, voteTx);
+      if (transactionId) {
+        dispatch(setTransactionsStateAction(transactionId));
```

Comment:
This should also be called for `transactionId === null` because then we know that `lastSignAndPostResult` was a rejection